src/fidesops/util/oauth_util.py

```diff
@@ -76,6 +76,20 @@ async def verify_oauth_client(
     return client
 
 
+async def get_current_user(
+    security_scopes: SecurityScopes,
+    authorization: str = Security(oauth2_scheme),
+    db: Session = Depends(deps.get_db),
+) -> FidesopsUser:
+    """A wrapper around verify_oauth_client that returns that client's user if one exsits."""
+    client = await verify_oauth_client(
+        security_scopes=security_scopes,
+        authorization=authorization,
+        db=db,
+    )
+    return client.user
```

Comment:
FYI @TheAndrewJackson this is what I ended up implementing to restrict editing data for only one user, as opposed to using a new scope as I suggested before.

Reply:
This is interesting. When `Depends(deps.get_db)` is called within this function does it reuse the same database connection from the routes that call `get_current_user`?

For example `update_user_password` has both `db: Session = Depends(deps.get_db)` and `current_user: FidesopsUser = Depends(get_current_user)` which both call `deps.get_db`.

Reply:
This is probably minor but the route decorator is also calling `verify_oauth_client` with `Security(verify_oauth_client, scopes=[USER_UPDATE])`. It looks like this work is being duplicated a little. It's probably negligible on performance since these routes won't be called that much

Reply:
> When Depends(deps.get_db) is called within this function does it reuse the same database connection from the routes that call get_current_user?

Dependency injection will pull a new connection from the session returned by the `get_db_session` function, so they aren't guaranteed to be the same connection, just from the same connection pool.

> It looks like this work is being duplicated a little.

It is! I didn't want to omit that security dependency entirely for readability.